airflow/providers/google/cloud/sensors/gcs.py

```diff
@@ -126,6 +130,9 @@ def execute(self, context: Context) -> None:
                     ),
                     method_name="execute_complete",
                 )
+            else:
+                if self._exists:
+                    return "success"
```

Comment:
If it always success, then how are we going to benefit from this? Can you elaborate the use case here?

Reply:
Thanks for taking a look, I've updated the PR description (didn't realize this would get look at while in draft). Today, if an object already exists when the DAG is first run, the xcom value is not set. "success" is what's returned today by the deferred mode trigger - https://github.com/apache/airflow/blob/main/airflow/providers/google/cloud/triggers/gcs.py#L111. The goal of this fix is essentially the same as what was done for GCSObjectsWithPrefixExistenceSensor in #37634.